source/ftrack_connect_pipeline_qt/ui/utility/widget/definition_selector.py

```diff
@@ -284,13 +284,36 @@ def populate_definitions(self):
                         )
                     )
                 if asset_type_name and self._host_connection.context_id:
-                    for asset_version in self._host_connection.session.query(
+                    # Collect versions
+                    asset_versions = self._host_connection.session.query(
                         'AssetVersion where '
                         'task.id={} and asset.type.name="{}" order by date descending'.format(
                             self._host_connection.context_id,
                             asset_type_name,
                         )
+                    ).all()
+                    # Follow context incoming link
+                    for tcl in self._host_connection.session.query(
+                        'TypedContextLink where to_id is "{}"'.format(
+                            self._host_connection.context_id
+                        )
                     ):
+                        for (
+                            asset_version
+                        ) in self._host_connection.session.query(
+                            'AssetVersion where '
+                            'asset.parent.id={} and asset.type.name="{}" order by date descending'.format(
+                                tcl['from_id'],
+                                asset_type_name,
+                            )
+                        ).all():
+                            if not asset_version['id'] in [
+                                _asset_version['id']
+                                for _asset_version in asset_versions
+                            ]:
+                                asset_versions.append(asset_version)
+
+                    for asset_version in asset_versions:
```

Comment:
There is to much logic in the QT layer, I should be able to open the correct asset build without using qt, so the logic should,'t be here.

Reply:
Will add a TODO so we can revisit this later and move this logic out of QT.